siddhi-test-framework/src/main/java/io/siddhi/distribution/test/framework/SiddhiRunnerContainer.java

```diff
@@ -209,18 +210,42 @@ private int getStartupTimeoutSeconds() {
     @Override
     protected void waitUntilContainerStarted() {
         logger().info("Waiting for Siddhi Runner Container to start...");
-        retryUntilSuccess(getStartupTimeoutSeconds(), TimeUnit.SECONDS, () -> {
-            if (!isRunning()) {
-                throw new ContainerLaunchException("Siddhi Runner Container failed to start");
-            }
-            HTTPClient.HTTPResponseMessage httpResponseMessage = callHealthAPI();
-            if (httpResponseMessage.getResponseCode() == 200) {
-                logger().info("Siddhi Runner Health API reached successfully.");
-                return null;
-            } else {
-                throw new ConnectException("Failed to connect with the Siddhi Runner health API");
+        if (SIDDHI_APPS != null) {
+            String[] siddhiApps = SIDDHI_APPS.list();
+            for (String siddhiApp : siddhiApps) {
+                String fileName = siddhiApp.substring(0, siddhiApp.length() - 7);
+                retryUntilSuccess(getStartupTimeoutSeconds(), TimeUnit.SECONDS, () -> {
```

Comment:
Seems this is repeating use a common function

Reply:
Fixed in https://github.com/siddhi-io/distribution/commit/684ff2789b8193d4b3820859614a6de441dbe914